### ISSUE TITLE
Avoiding ambiguity when the second-to-last author is different

### ISSUE THREAD

**Issue by rmolina** [plk/biblatex-apa#183]
Please consider the following MWE:

```latex
\documentclass{article}
\usepackage[style=apa]{biblatex}
\addbibresource{\jobname.bib}

\begin{filecontents}[force]{\jobname.bib}
@BOOK{One,
        AUTHOR = {Author Red and Author Green and Author Black and Author Blue},
        TITLE = {Title One},
        DATE = {2022},
        PUBLISHER = {Publisher},
}
@BOOK{Two,
        AUTHOR = {Author Red and Author Green and Author White and Author Blue},
        TITLE = {Title Two},
        DATE = {2022},
        PUBLISHER = {Publisher},
}
@BOOK{Three,
        AUTHOR = {Author Red and Author Green and Author Blue},
        TITLE = {Title Three},
        DATE = {2022},
        PUBLISHER = {Publisher},
}
\end{filecontents}

\begin{document}
\parencite{One,Two,Three}
\printbibliography
\end{document}
```

This code results in the citation `(Red, Green, Black, et al., 2022; Red, Green, & Blue, 2022; Red, Green, White, et al., 2022)`. I believe this should be `(Red, Green, Black, & Blue, 2022; Red, Green, & Blue, 2022; Red, Green, White, & Blue, 2022)`, since the Publication Manual  (Section 8.18) says: _Because “et al.” is plural (meaning “and others”), it cannot stand for only one name._

**Comment by plk** [comment]
Which version of the style are you using? I get the results you expect when I run this MWE. This case is explicitly mentioned with other additions too in the examples file distributed with the style.

**Comment by moewew** [comment]
FWIW I get

> (Red, Green, Black, et al., 2022; Red, Green, & Blue, 2022; Red, Green, White, et al., 2022)

(which I take is not the expected output) with the current release version, on Overleaf and with the current GitHub versions of `biblatex` and `biblatex-apa`.

**Comment by plk** [comment]
I think some wrinkle in this was fixed since the last release and so I am using the DEV versions ...

**Comment by moewew** [comment]
Very odd. I just grabbed `biblatex-apa` from `master`, `biblatex` from `dev` and the Biber binary from Sourceforge and got

>  (Red, Green, Black, et al., 2022; Red, Green, & Blue, 2022; Red, Green, White, et al., 2022)

The SF Biber binary is from April. Is yours more recent?

**Comment by rmolina** [comment]
Sorry about my late reply. I should have mentioned I was compiling the MWE using an up-to-date TeX Live 2022.

**Comment by plk** [comment]
Should be fixed in git - @moewew, I think I retired the code for this from the APA style a while ago as it was integrated into biblatex but it seems that the generic `\blx@namesetup` doesn't actually handle the case when `uniquelist` is one less than `listtotal`. For the fix I've overridden `\blx@namesetup` in the APA style but I wonder if this shouldn't be a biblatex fix as the plurality of "et al" is probably a general given. However, people might not expect this change, even though it's not a particularly common case?

**Comment by moewew** [comment]
I agree that in principle "et al." should really be plural. But I tested
```latex
\documentclass{article}
\usepackage[style=authoryear]{biblatex}
\addbibresource{\jobname.bib}
\addbibresource{biblatex-examples.bib}

\begin{filecontents}{\jobname.bib}
@BOOK{One,
        AUTHOR = {Author Red and Author Green and Author Black and Author Blue},
        TITLE = {Title One},
        DATE = {2022},
        PUBLISHER = {Publisher},
}
@BOOK{Two,
        AUTHOR = {Author Red and Author Green and Author White and Author Blue},
        TITLE = {Title Two},
        DATE = {2022},
        PUBLISHER = {Publisher},
}
@BOOK{Three,
        AUTHOR = {Author Red and Author Green and Author Blue},
        TITLE = {Title Three},
        DATE = {2022},
        PUBLISHER = {Publisher},
}
\end{filecontents}

\begin{document}
\parencite{One,Two,Three,companion,aksin}

\parencite{One,Two,Three,companion,aksin}

\printbibliography
\end{document}
```
with the standard `authoryear` style as far back as Overleaf would allow me and it always "et al."d the single author Blue.

So if something changed here, it must have been a very long time ago.

I'm all for a change on the `biblatex` side (even more so because I find it unfortunate that a style such as `biblatex-apa` has to hack internal macros like this), but then we should give people an option to switch back to the old behaviour. (Or if we want to be more careful, have an option to turn on the new behaviour.)

**Comment by plk** [comment]
Have a look at `biblatex` dev and `biblatex-apa` latest master branch commit. Moved this into core with a new "pluralothers" option, default false to retain backwards compat.

**Comment by moewew** [comment]
Looks good so far, but I was wondering if this might not need support on the Biber side. Consider the following example
```latex
\documentclass{article}
\usepackage[style=authoryear, uniquelist=false, minnames=3, pluralothers=true]{biblatex}
\addbibresource{\jobname.bib}
\addbibresource{biblatex-examples.bib}

\begin{filecontents}{\jobname.bib}
@BOOK{One,
        AUTHOR = {Author Red and Author Green and Author Blue and Author Black},
        TITLE = {Title One},
        DATE = {2022},
        PUBLISHER = {Publisher},
}
@BOOK{Two,
        AUTHOR = {Author Red and Author Green and Author Blue and Author White},
        TITLE = {Title Two},
        DATE = {2022},
        PUBLISHER = {Publisher},
}
\end{filecontents}

\begin{document}
\parencite{One,Two}

\parencite{One,Two}

\printbibliography
\end{document}
```
> (Red, Green, Blue, and Black 2022a; Red, Green, Blue, and White 2022b)

the disambiguation letters make sense from Biber's point of view, where the output would be

> (Red, Green, Blue, et al. 2022a; Red, Green, Blue, et al. 2022b)

but if we `pluralothers=true` we actually show the name that would disambiguate the name lists.

**Comment by plk** [comment]
Just updated biber DEV - should cope with this now - well spotted
